# Connecting to class database server from Db

# Question 1

In [9]:
import pandas as pd
from sqlalchemy import create_engine

def create_engine_conn():
    engine = create_engine("mysql+pymysql://infscompany:yeadata@128.198.162.191/company")
    return engine

In [10]:
# Check the Tables
engine = create_engine_conn()

tables_query = "SHOW TABLES"
tables_df = pd.read_sql_query(tables_query, engine)
print(tables_df)

    Tables_in_company
0         departments
1        dist_centers
2  expense_allocation
3        expense_type
4            expenses
5               sales
6          store_dept
7      store_turnover
8     store_type_desc
9              stores


In [11]:
dept_info = "SELECT * from departments WHERE DeptDescription LIKE '%%Media%%'"
df = pd.read_sql_query(dept_info, engine)
print(df)

   Dept        DeptDescription
0     5  Media and Gaming Cafe
1    55       Media and Gaming


# Retrive the data from tables

### Get all of the data from both the Sales and the Department table

In [12]:
sales_info = "SELECT * FROM sales"
sales_df = pd.read_sql_query(sales_info, engine)

dept_info = "SELECT * FROM departments"
dept_df = pd.read_sql_query(dept_info, engine)

merged_df = sales_df.merge(dept_df, on='Dept')

print(merged_df.head())

   PK  Store  Dept   SalesDate  Weekly_Sales  IsHoliday    DeptDescription
0   1    1.0   1.0  2019-02-05    25173.7450          0  Candy and Tobacco
1   2    1.0   1.0  2019-02-12    46499.8849          1  Candy and Tobacco
2   3    1.0   1.0  2019-02-19    42011.5055          0  Candy and Tobacco
3   4    1.0   1.0  2019-02-26    19597.5754          0  Candy and Tobacco
4   5    1.0   1.0  2019-03-05    22046.1790          0  Candy and Tobacco


# Show the weekly sales of the Media and Gaming Department

In [13]:
media_gaming_df = merged_df[merged_df['DeptDescription'] == 'Media and Gaming']
print(media_gaming_df)

            PK  Store  Dept   SalesDate  Weekly_Sales  IsHoliday  \
4569      4670    1.0  55.0  2019-02-05    36485.0653          0   
4570      4671    1.0  55.0  2019-02-12    32438.9548          1   
4571      4672    1.0  55.0  2019-02-19    29883.5780          0   
4572      4673    1.0  55.0  2019-02-26    28687.8077          0   
4573      4674    1.0  55.0  2019-03-05    27678.1087          0   
...        ...    ...   ...         ...           ...        ...   
366830  372414   33.0  55.0  2021-04-07        0.0000          0   
366917  372502    5.0  55.0  2021-04-07        0.0000          0   
366985  372570   32.0  55.0  2021-04-07     4495.9485          0   
367102  372692   46.0  55.0  2021-06-23        0.0000          0   
367178  372770   46.0  55.0  2021-06-30        0.0000          0   

         DeptDescription  
4569    Media and Gaming  
4570    Media and Gaming  
4571    Media and Gaming  
4572    Media and Gaming  
4573    Media and Gaming  
...                  

# Show the monthly sales

In [14]:
media_gaming_df['SalesDate'] = pd.to_datetime(media_gaming_df['SalesDate'])

media_gaming_df['Month'] = media_gaming_df['SalesDate'].dt.to_period('M') # thsi is creating the month column

monthly_sales = media_gaming_df.groupby('Month')['Weekly_Sales'].sum()

print(monthly_sales)

Month
2019-02    3.981645e+06
2019-03    3.312521e+06
2019-04    3.443901e+06
2019-05    2.422349e+06
2019-06    2.597272e+06
2019-07    3.052202e+06
2019-08    2.280960e+06
2019-09    2.276538e+06
2019-10    2.974144e+06
2019-11    5.440284e+06
2019-12    9.601961e+06
2020-01    2.360558e+06
2020-02    3.113560e+06
2020-03    2.856571e+06
2020-04    3.198303e+06
2020-05    2.025284e+06
2020-06    2.143700e+06
2020-07    2.529802e+06
2020-08    1.785908e+06
2020-09    2.707064e+06
2020-10    2.408672e+06
2020-11    5.884867e+06
2020-12    8.151055e+06
2021-01    6.722087e+06
2021-02    6.676937e+06
2021-03    8.097117e+06
2021-04    9.149297e+05
2021-05    9.076036e+05
2021-06    1.161671e+06
Freq: M, Name: Weekly_Sales, dtype: float64


/var/folders/5b/j6cfk3gs0ds6y6nzsqdg7ftw0000gn/T/ipykernel_54924/3583498406.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  media_gaming_df['SalesDate'] = pd.to_datetime(media_gaming_df['SalesDate'])
/var/folders/5b/j6cfk3gs0ds6y6nzsqdg7ftw0000gn/T/ipykernel_54924/3583498406.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  media_gaming_df['Month'] = media_gaming_df['SalesDate'].dt.to_period('M') # thsi is creating the month column


In [15]:
# Monthly Sales in Millions 

monthly_sales_million = monthly_sales / 1_000_000
print(monthly_sales_million)

Month
2019-02    3.981645
2019-03    3.312521
2019-04    3.443901
2019-05    2.422349
2019-06    2.597272
2019-07    3.052202
2019-08    2.280960
2019-09    2.276538
2019-10    2.974144
2019-11    5.440284
2019-12    9.601961
2020-01    2.360558
2020-02    3.113560
2020-03    2.856571
2020-04    3.198303
2020-05    2.025284
2020-06    2.143700
2020-07    2.529802
2020-08    1.785908
2020-09    2.707064
2020-10    2.408672
2020-11    5.884867
2020-12    8.151055
2021-01    6.722087
2021-02    6.676937
2021-03    8.097117
2021-04    0.914930
2021-05    0.907604
2021-06    1.161671
Freq: M, Name: Weekly_Sales, dtype: float64


# Find the expenses for Media and Gaming

In [16]:
query = """
select ea.ExpenseDate, ea.AllocatedAmount, ea.Store, ea.Dept
FROM expense_allocation ea
WHERE ea.Dept = 55
"""

expense_df = pd.read_sql_query(query, engine)

print(expense_df)

      ExpenseDate  AllocatedAmount  Store  Dept
0      2019-02-01       21607.9902    1.0  55.0
1      2019-03-01       97917.5800    1.0  55.0
2      2019-04-01       98113.4152    1.0  55.0
3      2019-05-01       98309.6420    1.0  55.0
4      2019-06-01       98506.2613    1.0  55.0
...           ...              ...    ...   ...
13061  2021-06-01        3686.4044   46.0  55.0
13062  2021-06-01       25202.4449   46.0  55.0
13063  2021-06-01        4319.8034   46.0  55.0
13064  2021-06-01       10395.5868   46.0  55.0
13065  2021-06-01        9701.2885   46.0  55.0

[13066 rows x 4 columns]


In [17]:
# Calculate expenses by month

expense_df['ExpenseDate'] = pd.to_datetime(expense_df['ExpenseDate'])

expense_df['Month'] = expense_df['ExpenseDate'].dt.to_period('M')

monthly_expense = expense_df.groupby('Month')['AllocatedAmount'].sum()

monthly_expense_million = monthly_expense / 1_000_000

print(monthly_expense_million)

Month
2019-02    16.295911
2019-03    20.811393
2019-04    20.859051
2019-05    20.902128
2019-06    20.013404
2019-07    16.244118
2019-08    15.110532
2019-09    15.522407
2019-10    15.583526
2019-11    15.622663
2019-12    15.671641
2020-01    14.258563
2020-02    10.421646
2020-03     9.943139
2020-04     9.703978
2020-05     9.114322
2020-06     9.762002
2020-07    12.272456
2020-08    12.283949
2020-09    12.303851
2020-10    12.327488
2020-11    12.363416
2020-12    12.701141
2021-01    12.661233
2021-02    11.078377
2021-03    11.352020
2021-04     2.180876
2021-05     1.798975
2021-06     1.887978
Freq: M, Name: AllocatedAmount, dtype: float64


# Caluclate Profit

In [18]:
monthly_profit_million = monthly_sales_million - monthly_expense_million

print(monthly_profit_million)

Month
2019-02   -12.314266
2019-03   -17.498872
2019-04   -17.415150
2019-05   -18.479779
2019-06   -17.416132
2019-07   -13.191916
2019-08   -12.829573
2019-09   -13.245869
2019-10   -12.609382
2019-11   -10.182379
2019-12    -6.069679
2020-01   -11.898006
2020-02    -7.308086
2020-03    -7.086568
2020-04    -6.505674
2020-05    -7.089038
2020-06    -7.618302
2020-07    -9.742654
2020-08   -10.498042
2020-09    -9.596787
2020-10    -9.918816
2020-11    -6.478548
2020-12    -4.550086
2021-01    -5.939146
2021-02    -4.401440
2021-03    -3.254904
2021-04    -1.265946
2021-05    -0.891371
2021-06    -0.726307
Freq: M, dtype: float64


In [19]:
sales_sum = monthly_sales_million.sum()
expense_sum = monthly_expense_million.sum()

print(sales_sum)
print(expense_sum)

105.02946756359998
371.0521847443


In [20]:
net_loss = sales_sum - expense_sum

print(net_loss)

-266.02271718070006


In [27]:
summary_df = pd.DataFrame({
    'Sales ($M)': monthly_sales_million,
    'Expenses ($M)': monthly_expense_million,
    'Profti/Loss ($M)': monthly_profit_million
})

print(summary_df)

print(f'Total sales: {sales_sum}')
print(f'Total expenses: {expense_sum}')
print(f'Total Profit/Loss: {net_loss}')

         Sales ($M)  Expenses ($M)  Profti/Loss ($M)
Month                                               
2019-02    3.981645      16.295911        -12.314266
2019-03    3.312521      20.811393        -17.498872
2019-04    3.443901      20.859051        -17.415150
2019-05    2.422349      20.902128        -18.479779
2019-06    2.597272      20.013404        -17.416132
2019-07    3.052202      16.244118        -13.191916
2019-08    2.280960      15.110532        -12.829573
2019-09    2.276538      15.522407        -13.245869
2019-10    2.974144      15.583526        -12.609382
2019-11    5.440284      15.622663        -10.182379
2019-12    9.601961      15.671641         -6.069679
2020-01    2.360558      14.258563        -11.898006
2020-02    3.113560      10.421646         -7.308086
2020-03    2.856571       9.943139         -7.086568
2020-04    3.198303       9.703978         -6.505674
2020-05    2.025284       9.114322         -7.089038
2020-06    2.143700       9.762002         -7.

# Question 2

The Media and Gaming department is showing a loss much greater than the sales it is producing. Sales currently sit at a total of 105 million and expenses sit at around 371 million. 

Sales are being calculated by the sales of products in this department, but expenses are being caluclated by sqaure footage and operating costs like the rent, utilities, etc. 

The department is making good sales, but is unable to cover the overhead costs. 

# Question 3

My recommendation is to not invest additional captial in video games at this time:

We are showing a net loss of around $266 million

The department cannot cover the overhead costs currently where sales are at


In order for the company to reconsider investing in this department we would need to:

Reduce overhead costs or restructure allocation of expesnes 

Develop a plan to become more profitable or a postive contribution margin

Identify specific areas within the department that would help significantly increase sales